pkgs/development/tools/aspect-cli/default.nix

```diff
@@ -0,0 +1,40 @@
+{ stdenv, lib, fetchurl }:
+let
+  platform = {
+    x86_64-linux = "linux_amd64";
+    aarch64-linux = "linux_arm64";
+    x86_64-darwin = "darwin_amd64";
+    aarch64-darwin = "darwin_arm64";
+  }.${stdenv.hostPlatform.system} or (throw "unsupported system ${stdenv.hostPlatform.system}");
+
+  sha256 = {
+    x86_64-linux = "sha256-ZCmy+4jyEzrto8R9kE7tNDoy9Pq/lVjkR6dZHMyK5AE=";
+    aarch64-linux = "sha256-qEl9ch4qNNtNniyZM56dr7jtxUD8ZqKq1mxbkl7ISfw=";
+    x86_64-darwin = "sha256-vsBOYhvXmEg7tOIxkDigLOu9G2jUjqg9rB4B/L7bd64=";
+    aarch64-darwin = "sha256-YI2kOHkQ1FF/Y8EWF/H8dPwnrSkjwqWfOtDrd3gdVtY=";
+  }.${stdenv.hostPlatform.system} or (throw "unsupported system ${stdenv.hostPlatform.system}");
+in
+stdenv.mkDerivation rec {
+  pname = "aspect-cli";
+  version = "5.5.4";
+
+  src = fetchurl {
+    inherit sha256;
+    url = "https://github.com/aspect-build/aspect-cli/releases/download/${version}/aspect-${platform}";
+  };
+
+  phases = [ "installPhase" ];
```

Comment:
Disabling phases like this is [kinda illegal](https://nixos.org/manual/nixpkgs/stable/#reviewing-contributions-new-packages)

Reply:
I see, I have seen this pattern around and thought it was ok to use (e.g. the very [mkShell derivation](https://github.com/NixOS/nixpkgs/blob/da3790467235e8b37dea585bb94da4885d495806/pkgs/build-support/mkshell/default.nix#L44)).

What would you have me do instead?

Reply:
I think the rules are different for the build-support stuff.

If you don't need certain phases, you can usually just ignore them, or optionally disable them with attrs like `dontUnpack`, `dontConfigure`, `dontBuild`, etc. if they are causing errors. You can read more about what's available here: https://nixos.org/manual/nixpkgs/stable/#sec-stdenv-phases

Reply:
Done and tested on a local flake, seems to be working. Can you take a look again? 🙌🏻